### PR TITLE
feat(autogpt_server): REST API and WS API now use env vars for host and port.

### PR DESCRIPTION
**Pull request by twinsant**
### Background

Previous versions used hard-coded settings for the server's host and port, which was inconvenient during development

### Changes 🏗️

Added host and port settings to the .env file

___

### **PR Type**
enhancement, configuration changes

___

### **Description**
- Updated the REST API and WS API to use environment variables for configuring the host and port, enhancing flexibility and ease of configuration.
- Modified `.env.example` to include example environment variables for the REST and WS API host and port settings.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>rest_api.py</strong><dd><code>Use environment variables for REST API host and port</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

rnd/autogpt_server/autogpt_server/server/rest_api.py

<li>Added import for <code>os</code> module.<br> <li> Updated <code>uvicorn.run</code> to use environment variables for host and port.<br>

</details>

  </td>
  <td><a href="https://github.com/Significant-Gravitas/AutoGPT/pull/7970/files#diff-b2f55efb9917123e95d280bec2c6bf3ad81ee9398a39a756e7974643a3ad2021">+6/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>ws_app.py</strong><dd><code>Use environment variables for WS API host and port</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

rnd/autogpt_server/autogpt_server/ws_app.py

<li>Added import for <code>os</code> module.<br> <li> Updated <code>uvicorn.run</code> to use environment variables for host and port.<br>

</details>

  </td>
  <td><a href="https://github.com/Significant-Gravitas/AutoGPT/pull/7970/files#diff-6f2a9695bf4fe8f895608d84f0eee001045a1b81c04f9396b3d110c5bb648552">+6/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></td></tr><tr><td><strong>Configuration changes</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>.env.example</strong><dd><code>Add example environment variables for server configuration</code></dd></summary>
<hr>

rnd/autogpt_server/.env.example

<li>Added example environment variables for REST and WS API host and port.<br> <br>

</details>

  </td>
  <td><a href="https://github.com/Significant-Gravitas/AutoGPT/pull/7970/files#diff-9dd1e3bfac62caa79a5a78c43140da9ea45d0d3f4d20f14b1fe894d6c40ce0af">+8/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

File: rnd/autogpt_server/.env.example [hunk 1]
```diff
@@ -6,3 +6,11 @@ DATABASE_URL="postgresql://${DB_USER}:${DB_PASS}@localhost:${DB_PORT}/${DB_NAME}
 PRISMA_SCHEMA="postgres/schema.prisma"
 ENABLE_AUTH="false"
 APP_ENV="local"
+
+# REST API
+AP_SERVER_HOST="0.0.0.0"
```
Comment (aarushik93) [inline rnd/autogpt_server/.env.example]: can you add these to the docker compose as well please

Comment (twinsant) [reply rnd/autogpt_server/.env.example]: I will

**Review by majdyz — APPROVED** [review]
Thanks for adding this change!

**Comment by Swiftyos** [comment]
Sorry we should have merged it for you once it was approved.

Could you resolve the merge conflicts and we will merge it in.

**Pull request closed without merging**